test/test-mongocrypt-marking.c

```diff
@@ -1678,9 +1694,68 @@ static void test_mc_marking_to_ciphertext_fle2_text_search(_mongocrypt_tester_t
     }
 }
 
+static void test_ciphertext_len_steps_fle2_text_search(_mongocrypt_tester_t *tester) {
+    const char *markingJSONFormat = RAW_STRING({
+        't' : 1,
+        'a' : 4,
+        'v' : {
+            'v' : "%s",
+            'casef' : false,
+            'diacf' : false,
+            'suffix' : {'ub' : {'$numberInt' : '2'}, 'lb' : {'$numberInt' : '1'}}
+        },
+        'cm' : {'$numberLong' : '2'}
+    });
+    size_t last_len = 0;
+    mongocrypt_binary_t *cmd = TEST_FILE("./test/example/cmd.json");
+    mongocrypt_binary_t *key_file = TEST_BIN(16);
+    mongocrypt_binary_t *ki = TEST_BIN(16);
+    mongocrypt_binary_t *ku = TEST_BIN(16);
+
+    for (size_t strlen = 0; strlen < 256; strlen++) {
```

Comment:
```suggestion
    for (size_t str_len = 0; str_len < 256; str_len++) {
```

Suggest renaming to avoid conflicting with `strlen()` function.